### PR TITLE
Add logging to build_manager.set_environment_vars().

### PR DESCRIPTION
**Pull request by letitz**
This should surface errors earlier and provide more context to debug failures.
Specifically, I am hoping it will help diagnose https://crbug.com/1497929.

### CODE REVIEW

File: src/clusterfuzz/_internal/build_management/build_manager.py [hunk 1]
```diff
@@ -344,11 +344,28 @@ def set_environment_vars(search_directories, app_path='APP_PATH',
...
   # Chromium specific folder to ignore.
   initialexe_folder_path = f'{os.path.sep}initialexe'
 
+  logs.log('\n'.join([
+      'Walking build directory to find files and set environment variables.',
+      f'  Environment prefix: {env_prefix!r}',
+      f'  App path environment variable name: {app_path!r}',
+      f'  App name: {app_name!r}',
+      f'  LLVM symbolizer file name: {llvm_symbolizer_filename!r}',
+      f'  Use default LLVM symbolizer: {use_default_llvm_symbolizer}',
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/build_management/build_manager.py:359]: nit: not sure we need leading spaces.

**Review by jonathanmetzman — COMMENTED** [review]
DO you want to add bad build marking or in another PR?

File: src/clusterfuzz/_internal/build_management/build_manager.py (hunk 1 above)
Comment (letitz) [reply src/clusterfuzz/_internal/build_management/build_manager.py:359]: Will remove.

**Comment by letitz** [comment]
I'll mark the build as bad in a followup PR.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Comment by jonathanmetzman** [comment]
Nice this catches the error: https://pantheon.corp.google.com/errors/detail/CLuEsY-G8c3aaA;time=P30D?project=google.com:clusterfuzz&utm_source=error-reporting-notification&utm_medium=email&utm_content=new-error&e=-13802955&mods=logs_tg_prod

**Pull request merged**